src/LDLFactorizations.jl

```diff
@@ -192,6 +199,13 @@ function ldl_numeric_upper!(n, Ap, Ai, Ax, Cp, Ci, Lp, parent, Lnz, Li, Lx, D, Y
       Lnz[i] += 1
       top += 1
     end
+    if dynamic_regul && abs(D[k]) < eps(T) * dynamic_dict[:Amax]
+      if P[k] <= dynamic_dict[:n_d]
+        D[k] += dynamic_dict[:r1]
+      else
+        D[k] += dynamic_dict[:r2]
```

Comment:
Should this be - instead of +? Or are you expecting r2 < 0?

Reply:
In my interior point code I expect r1<0 and r2>0

Reply:
Ok I see. But it's good to have the flexibility to specify any r1 and r2.